src/components/CommentAdder.jsx

```diff
@@ -0,0 +1,79 @@
+import { useContext, useState } from 'react';
+import { UserContext } from '../contexts/userContext';
+import { postComment } from '../utils/utils';
+
+const CommentAdder = ({ article_id, setComments, setCommentCount }) => {
+  const { user } = useContext(UserContext);
+  const [commentInput, setCommentInput] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [success, setSuccess] = useState(false);
+
+  const inputHandler = event => {
+    setCommentInput(event.target.value);
+    setError(false);
+    setSuccess(false);
+  };
+
+  const submitHandler = event => {
+    event.preventDefault();
+    postComment(article_id, user, commentInput)
+      .then(newComment => {
+        setIsLoading(true);
+
+        setTimeout(() => {
+          setIsLoading(false);
+          setSuccess(true);
+          setCommentCount(currCount => {
+            return ++currCount;
+          });
+          setComments(currComments => {
+            return [newComment, ...currComments];
+          });
+          setTimeout(() => {
+            setSuccess(false);
+          }, 2000);
+        }, 1000);
+      })
+      .catch(err => {
+        setError(true);
+      });
+
+    setCommentInput('');
+  };
+
+  if (user !== '') {
+    return (
+      <form id="comment-form" onSubmit={submitHandler}>
+        <textarea
+          name=""
+          id=""
+          cols="100"
+          rows="10"
+          required
+          minLength="5"
+          maxLength="400"
+          onChange={inputHandler}
+          value={commentInput}
+        ></textarea>
+        <button disabled={isLoading ? true : false}>
```

Comment:
great that you've prevented user pressing this button while posting is in progress 👍